### DISCUSSION TITLE
Error mapping ValueTuple to ValueTuple

### DISCUSSION THREAD

**Discussion by mm3141** [LuckyPennySoftware/AutoMapper#4068] [NA]
### Source/destination types

```csharp
		public class U
		{
			public int A { get; set; }
			public int S { get; set; }
		}

		public class V
		{
			public int B { get; set; }
		}

		public class X
		{
			public int A { get; set; }
			public int B { get; set; }
		}

		public record Y(U u, V v);
```

### Mapping configuration

```csharp
		sealed class MappingProfile : Profile
		{
			public MappingProfile()
			{
				CreateMap<Y, X>(MemberList.Destination)
					.ForMember(d => d.A, o => o.MapFrom(s => s.u.A))
					.ForMember(d => d.B, o => o.MapFrom(s => s.v.B));

				// with a source record, it is working
				CreateMap<Y, (X x, int s)>()
					.ForMember(d => d.x, o => o.MapFrom(s => s))
					.ForMember(d => d.s, o => o.MapFrom(s => s.u.S));

				CreateMap<(U u, V v), X>(MemberList.Destination)
					.ForMember(d => d.A, o => o.MapFrom(s => s.u.A))
					.ForMember(d => d.B, o => o.MapFrom(s => s.v.B));

				// with a source ValueTuple, it is not working
				CreateMap<(U u, V v), (X x, int s)>()
					.ForMember(d => d.x, o => o.MapFrom(s => s))
					.ForMember(d => d.s, o => o.MapFrom(s => s.u.S));
			}
		}
```

### Expected behavior

Everything is mapped

### Actual behavior
Mapping fails with an exception like this:
```
AutoMapper.AutoMapperConfigurationException
The following member on System.ValueTuple`2[[Tests+X, Tests, Version=0.14.0.0, Culture=neutral, PublicKeyToken=6a37e968aab193a7],[System.Int32, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]] cannot be mapped: 
	item1 
Add a custom mapping expression, ignore, add a custom resolver, or modify the destination type System.ValueTuple`2[[Tests+X, Tests, Version=0.14.0.0, Culture=neutral, PublicKeyToken=6a37e968aab193a7],[System.Int32, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].
Context:
	Mapping to member item1 from System.ValueTuple`2[[Tests+U, Tests, Version=0.14.0.0, Culture=neutral, PublicKeyToken=6a37e968aab193a7],[Tests+V, Tests, Version=0.14.0.0, Culture=neutral, PublicKeyToken=6a37e968aab193a7]] to System.ValueTuple`2[[Tests+X, Tests, Version=0.14.0.0, Culture=neutral, PublicKeyToken=6a37e968aab193a7],[System.Int32, System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]]
```

### Steps to reproduce

```csharp
		[Fact]
		public void Automapper_ValueTupleTest()
		{
			new MapperConfiguration( x => x.AddProfile(new MappingProfile())).AssertConfigurationIsValid();
		}
```

**Comment by lbargaoanu** [comment]
Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build.

**Reply by mm3141** [reply]
Same error with 11.0.2-alpha.0.12

**Reply by lbargaoanu** [reply]
Definitely not the same :) And the error tells you exactly what's wrong.
But that aside, I don't think this kind of mapping makes much sense. You can name your value tuples so everything matches (AutoMapper, you know), have proper DTOs, write some one line type converters. Each of those options would work a lot better than what you're trying to do here. `IncludeMembers` seems to apply too.

**Reply by mm3141** [reply]
Well, this is a MWE, the real thing uses `.ProjectTo` and will need multiple related entities as input, and should yield some proper DTO and some additional information needed at the same time. That may explain the design looking weird at the first glance.

I have double-checked the assembly version, the error message does not give me more information on what is going wrong here. All mappings are explicit in the example to make sure that there is no misbehavior from the automatic part. Mapping a class/record to a ValueTuple (named or not does not matter) is working fine, mapping a ValueTuple to some class/record is working fine, too, only ValueTuple on both sides does not work, even when explicitly specifying all the mappings.

**Reply by lbargaoanu** [reply]
This is not a support site, you just need to fix your code, but again, I really don't see the point of these mappings. Once you figure it out, you can open a PR if you have ideas for improvements.
